src/portfolio.py

```diff
@@ -0,0 +1,297 @@
+
+############################################################################
+### GSCO 2024 - PORTFOLIO AND STRATEGY CLASSES
+############################################################################
+
+# --------------------------------------------------------------------------
+# Cyril Bachelard
+# This version:     13.06.2024
+# First version:    13.06.2024
+# --------------------------------------------------------------------------
+
+
+import pandas as pd
+import numpy as np
+from typing import Callable, Dict, List
+
+
+
+# --------------------------------------------------------------------------
+def floating_weights(X, w, start_date, end_date, rescale = True):

```

Comment:
Shouldn't it be a class member of Portfolio class?

Reply:
I think this function is more of a helper function as it takes any weights and return series X to recompute the weights. It is called by another float_weights method of Portfolio